Game.java

```diff
@@ -35,27 +31,37 @@ public String showBoard(Board board) {
     }
 
     public Board playerMakesMove() {
-        if (getTurn() == 'o' && gameMode.equals("pvc")) {
-            computerPlayer = new ComputerPlayer();
-            board = board.playerMakesMove(getTurn(), computerPlayer.nextMove(board));
-        } else if(gameMode.equals("cvc")) {
-            computerPlayer = new ComputerPlayer();
-            board = board.playerMakesMove(getTurn(), computerPlayer.nextMove(board));
-        } else {
-            int position = scanner.nextInt();
-            if (!board.availablePosition(position - 1)) return board;
-            board = board.playerMakesMove(getTurn(), position - 1);
+        int position = playerActive.nextMove(board);
+        if (!board.availablePosition(position - 1)) {
+            print("Invalid Position");
```

Comment:
When you separate out the UI concerns, have a think whether Game needs to know about the position being one-indexed?